### ISSUE TITLE
Implement "virtual spaces"

### ISSUE THREAD

**Issue by nippur72** [GoogleChrome/chromedeveditor#3922]
Most text editors have an option called "virtual spaces" by which the caret can go beyond the end of line. The pleasurable effect is that when you navigate up and down the caret isn't forced to the end of line and it's steady on its column value. 

User accustomed to this option find very hard to live without.

**Comment by The13thDoc** [comment]
You bring up an interesting point (in my opinion). I've never used that feature.

Though, the caret *does* return to the same point, if that line contains a character, when moving up and down, even to lines where it travels to the end of a line. Is that not accomplishing something similar?

**Comment by nippur72** [comment]
Yes, of course the caret returns to the same position, but still it's too "jumpy" IMO.

Virtual space helps you in some situations (e.g. editing near the EOLs) and gives you more freedom. 

Selection on multiple lines is also affected by virtual space, as selection is shown as a rectangular block rather than line-fragmented.
